### PR TITLE
PHP: Add Enum methods for converting to/from strings

### PR DESCRIPTION
**Pull request by michaelbausor**
This PR migrates the changes in https://github.com/protocolbuffers/protobuf/pull/4534 and rebases them onto master branch. cc @TeBoring @fiboknacky

### CODE REVIEW

**Comment by TeBoring** [comment]
It seems some tests are broken. Could you help take a look?

**Comment by michaelbausor** [comment]
@TeBoring updated to not use EnumTrait, PTAL

**Comment by TeBoring** [comment]
It seems the test is still broken. Basically, GPBUtil should also be avoided in generated code.

**Comment by TeBoring** [comment]
Doesn't simply expand the GPBUtil method in the generated code work?

**Comment by michaelbausor** [comment]
Yes, it does. I thought that using GPBUtil would work, but it seems not (I'll look into why my local testing caught the earlier error with EnumTrait, but not this one GPBUtil).

**Comment by michaelbausor** [comment]
Updated, PTAL

**Comment by TeBoring** [comment]
Still some broken tests. Please take a look.

File: php/tests/well_known_test.php [hunk 1]
```diff
@@ -392,4 +392,33 @@ public function testWrappers()
...
+            $this->assertSame($v, $class::value($k));
+        }
+    }
+
+    public function enumNameValueConversionDataProvider()
+    {
+        return [
+            ['\Google\Protobuf\Field\Cardinality'],
+            ['\Google\Protobuf\Field\Kind'],
+            ['\Google\Protobuf\Internal\FieldDescriptorProto\Label'],
```
Comment (TeBoring) [inline php/tests/well_known_test.php]: These messages under Google\Protobuf\Internal are internal only and don't have implementation in c extension.
We don't need to test them.

File: php/src/Google/Protobuf/Syntax.php [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 namespace Google\Protobuf;
 
+use UnexpectedValueException;
```
Comment (TeBoring) [inline php/src/Google/Protobuf/Syntax.php]: I got warning: The use statement with non-compound name 'UnexpectedValueException' has no effect
Remove that?

**Comment by TeBoring** [comment]
Implemented c extension in the last commit.

File: php/src/Google/Protobuf/Syntax.php (hunk 2 above)
Comment (michaelbausor) [reply php/src/Google/Protobuf/Syntax.php]: I found that I got this warning only for the no-namespace enum. Is that what you found? I have added a commit to remove the use statement from the no-namespace case.

Comment (TeBoring) [reply php/src/Google/Protobuf/Syntax.php]: yes. For NoNamespace enum

**Comment by TeBoring** [comment]
@michaelbausor Do you want to take another look?

**Comment by bshaffer** [comment]
Hooray! nice work guys!

**Comment by beccasaurus** [comment]
@michaelbausor et al – any update on when this will make it into the PHP protobuf library?

Tried using this today and there is no method `::name` on my enum class, presuming I'm calling it correctly.

**Comment by TeBoring** [comment]
3.7.0rc2 should already have this

**Pull request merged**